Assets/Mirror/Components/Discovery/NetworkDiscoveryHUD.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Mirror.Discovery
+{
+    [DisallowMultipleComponent]
+    [AddComponentMenu("Network/NetworkDiscoveryHUD")]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkDiscovery.html")]
+    [RequireComponent(typeof(NetworkDiscovery))]
+    public class NetworkDiscoveryHUD : MonoBehaviour
+    {
+        readonly Dictionary<long, ServerInfo> discoveredServers = new Dictionary<long, ServerInfo>();
+        Vector2 scrollViewPos = Vector2.zero;
+
+        public NetworkDiscovery networkDiscovery;
+
+#if UNITY_EDITOR
+        private void OnValidate()
+        {
+            if (networkDiscovery == null)
+                networkDiscovery = GetComponent<NetworkDiscovery>();
+        }
+#endif
+
+        void OnEnable()
+        {
+            NetworkDiscovery.OnServerFound += OnDiscoveredServer;
```

Comment:
```suggestion
            NetworkDiscovery.OnServerFound += OnDiscoveredServer;
            
            discoveredServers.Clear();
            networkDiscovery.StartDiscovery();
```
Automatically perform "Find Servers" once.